handler/parse.go

```diff
@@ -100,46 +101,69 @@ func (dt dataTree) mount(i, v []string) error {
 	return dt[i[0]].(dataTree).mount(i[1:], v)
 }
 
-func (dt dataTree) prepareNewDataNode(i, v []string) (bool, error) {
-	_, ok := dt[i[0]]
-	if !ok {
-		dt[i[0]] = make(dataTree)
-		return true, nil
+func prepareTreeNode[T dataTree | fileTree, V []string | []*FileUpload](
```

Comment:
Could you please put the args into 1 line and slightly update the `hasBranch` check in the following way (to be more clear I guess):

```go
func prepareTreeNode[T dataTree | fileTree, V []string | []*FileUpload](tree T, i []string, v V) (bool, error) {
	if _, ok := tree[i[0]]; !ok {
		tree[i[0]] = make(T)
		return false, nil
	}
....
```